File: lib/models/Product.dart
class Product {
  final int? id;
  final int? categoryId;
  final String? name;
  final double? price;
  final String? thumbnail;
  final String? description;
  final double? weight;

  Product({
    this.id,
    this.categoryId,
    this.name,
    this.price,
    this.thumbnail,
    this.description,
    this.weight,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      categoryId: json['category_id'],
      name: json['name'],
      price: double.tryParse(json['price']?.toString() ?? '0'), // Chuyển đổi chuỗi sang double
      thumbnail: json['thumbnail'],
      description: json['description'],
      weight: double.tryParse(json['weight']?.toString() ?? '0'), // Chuyển đổi chuỗi sang double
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'category_id': categoryId,
      'name': name,
      'price': price,
      'thumbnail': thumbnail,
      'description': description,
      'weight': weight,
    };
  }
}

File: BACKENDpHp/FRUZZIY/web_api/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Product extends Model
{
    use HasFactory;

    protected $fillable = [
        'category_id',
        'name',
        'price',
        'thumbnail',
        'description',
        'weight',
    ];

    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    public function warehouses()
    {
        return $this->hasOne(Warehouse::class);
    }
}
